src/mlpack/bindings/go/get_printable_type_impl.hpp

```diff
@@ -0,0 +1,151 @@
+    const typename boost::disable_if<data::HasSerialize<T>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "unknown";
+}
+
+template<>
+inline std::string GetPrintableType<int>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<int>>::type*,
+    const typename boost::disable_if<data::HasSerialize<int>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<int>>::type*,
+    const typename boost::disable_if<std::is_same<int,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "int";
+}
+
+template<>
+inline std::string GetPrintableType<double>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<double>>::type*,
+    const typename boost::disable_if<data::HasSerialize<double>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<double>>::type*,
+    const typename boost::disable_if<std::is_same<double,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "float64";
+}
+
+template<>
+inline std::string GetPrintableType<std::string>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<std::string>>::type*,
+    const typename boost::disable_if<data::HasSerialize<std::string>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<std::string>>::type*,
+    const typename boost::disable_if<std::is_same<std::string,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "string";
+}
+
+template<>
+inline std::string GetPrintableType<size_t>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<size_t>>::type*,
+    const typename boost::disable_if<data::HasSerialize<size_t>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<size_t>>::type*,
+    const typename boost::disable_if<std::is_same<size_t,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "float64";
+}
+
+template<>
+inline std::string GetPrintableType<bool>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<bool>>::type*,
+    const typename boost::disable_if<data::HasSerialize<bool>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<bool>>::type*,
+    const typename boost::disable_if<std::is_same<bool,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "bool";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& d,
+    const typename boost::enable_if<util::IsStdVector<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "list of " + GetPrintableType<typename T::value_type>(d) + "s";
```

Comment:
I want some help with all this documentation part. Sorry I didn't comment about this part.

Reply:
Definitely, no worries.  For `GetPrintableType()`, these are the types that show up in the "type" column of the documentation; e.g., the second column in the "input parameters" table here: http://mlpack.org/doc/mlpack-3.1.1/cli_documentation.html#gmm_generate
(You can try changing the language to Python in the upper left to see about those types too.)

Reply:
I mean I want some help about what must be the correct documentation here.

Reply:
Right, so basically taking a look through other Go documentation might be helpful here.  In Python, arrays are usually called lists, so this code would emit "list of ints" or "list of strings", which is what a Python user would expect to see.  For Go, I am not sure, but I imagine it would come out something like "array of strings" or "array of ints".  It seems like 'array' is a reasonable word to use based on this: https://tour.golang.org/moretypes/6

Anyway, let me know what you think.  We don't have to go with that if you think you have a more appropriate term.  The real key is to try and make mlpack's documentation for Go feel as Go-native as possible, even though mlpack isn't written in Go.

Reply:
Looks like this is handled in the latest revision, so I'll resolve the conversation. :+1: